Update dependencies and minor cleanup

XhmikosR opened the pull request.

XhmikosR committed:
Update dependencies.

XhmikosR committed:
Normalize package.json.

* Fix `npm test` on Windows
* Use https for the repository

XhmikosR committed:
Add missing semicolon.

jharding commented on README.md:
```diff
@@ -84,5 +86,5 @@ For more information on semantic versioning, please visit http://semver.org/.
 License
 -------
 
-Copyright (c) 2013 [Jake Harding](http://thejakeharding.com)  
```
Can you revert this change as it's needed to insert a newline.

XhmikosR replied:
I went with a clearer approach using `<br/>`

jharding commented on tasks/sed.js:
```diff
@@ -6,7 +6,7 @@
 module.exports = function(grunt) {
   var replace = require('replace')
-    , _ = grunt.util._
+    , _ = require('lodash')
```
What's gained from relying on lodash directly?

XhmikosR replied:
http://gruntjs.com/api/grunt.util#grunt.util._

XhmikosR committed:
Switch to "grunt-exec" for the test command.

Now everything is handled with Grunt.

Also add a default target while we at it.

XhmikosR commented:
I rebased the branch to keep only the package.json changes; I'll make a new PR for the rest after this is merged.

XhmikosR committed:
Update .gitignore.

XhmikosR committed:
Update README.md.

* Add dependency badges
* Tweak code highlighting

XhmikosR committed:
Move JSHint's config to .jshintrc.

XhmikosR committed:
Explicitly set the files to install.
